src/services/PersistenceService/index.ts

```diff
@@ -0,0 +1,60 @@
+import { BooleanT } from '../../utils/array'
+
+export const NAMESPACE_PS = 'ps'
+
+export const KEY_DATA_VERSION = `${NAMESPACE_PS}_dataVersion`
+
+export class PersistenceService {
+  oldDataVersion = this.get<string>(KEY_DATA_VERSION)
+
+  currentDataVersion = '1'
+
+  dataVersionChanged = this.oldDataVersion !== this.currentDataVersion
+
+  constructor() {
+    this.handleDataMigration()
+    this.set<string>(KEY_DATA_VERSION, this.currentDataVersion)
+  }
+
+  handleDataMigration() {
+    if (!this.dataVersionChanged) return
+
+    switch (this.oldDataVersion) {
+      case null:
+      case undefined:
+        localStorage.clear()
+        break
+
+      default:
+        break
+    }
+  }
+
+  get<T = unknown>(key: string): T | undefined
+  get<T = unknown>(key: string, defaultValue: T): T
+  get<T = unknown>(key: string, defaultValue?: T): T | undefined {
+    const jsonStr = localStorage.getItem(key)
+    if (!jsonStr) return defaultValue
+    return JSON.parse(jsonStr) as T
+  }
+
+  set<T = unknown>(key: string, value: T): T {
+    localStorage.setItem(key, JSON.stringify(value))
+    return value
+  }
+
+  remove(key: string): void {
+    localStorage.removeItem(key)
+  }
+
+  getAllKeys() {
+    return new Array(localStorage.length)
+      .fill(null)
+      .map((_, idx) => localStorage.key(idx))
```

Comment:
What's the difference from `Object.keys(localStorage)`

Reply:
I was thinking that this might be safer, because the meaning represented by Object.keys is not equal to getting localStorage keys.

For example:

```javascript
Object.keys(localStorage)
// ['cache_testnet_MinerVersionDistribution']

localStorage.setItem = () => console.log('set')

Object.keys(localStorage)
// ['setItem', 'cache_testnet_MinerVersionDistribution']

new Array(localStorage.length).fill(null).map((_, idx) => localStorage.key(idx))
// ['cache_testnet_MinerVersionDistribution']
```

Reply:
Got it